### PR TITLE
Spark Applications metrics collector: debug prints 

### PR DESCRIPTION
**Pull request by roi-granulate**

### CODE REVIEW

File: granulate_utils/metrics/spark.py [hunk 1]
```diff
@@ -121,8 +121,15 @@ def _get_standalone_apps(self) -> Dict[str, Tuple[str, str]]:
...
         metrics_json = rest_request_to_json(self._master_address, SPARK_MASTER_STATE_PATH)
         running_apps = {}
+        current_running_apps = []
 
-        for app in metrics_json.get("activeapps", []):
+        try:
+            current_running_apps = metrics_json["activeapps"]
+        except KeyError:
+            # Log the exception where the activeapps key is not found, and log the response
+            self._logger.exception("Could not find 'activeapps' in metrics_json", metrics_json=metrics_json)
```
Comment (Jongy) [inline granulate_utils/metrics/spark.py:130]: No need to log the exception backtrace itself (`logger.exception`) as we expect it. Just catch the KeyError and log a WARNING "No activeapps in metrics!"

File: granulate_utils/metrics/spark.py [hunk 2]
```diff
@@ -143,6 +150,11 @@ def _get_standalone_apps(self) -> Dict[str, Tuple[str, str]]:
                     pass
             except Exception:
                 self._logger.exception("Error was found while iterating applications.")
+        else:
+            if running_apps == {}:
+                self._logger.debug("running_apps is empty", metrics_json=metrics_json)
+            if current_running_apps == []:
+                self._logger.debug("current_running_apps is empty", metrics_json=metrics_json)
```
Comment (Jongy) [inline granulate_utils/metrics/spark.py:157]: Always log the size of running apps & current running apps, in a single log message.

File: granulate_utils/metrics/spark.py (hunk 1 above)
Comment (Jongy) [inline granulate_utils/metrics/spark.py:130]: No need to log the exception backtrace itself (`logger.exception`) as we expect it. Just catch the KeyError and log a WARNING "No activeapps in metrics!"

File: granulate_utils/metrics/spark.py [hunk 3]
```diff
@@ -265,11 +280,18 @@ def _spark_executor_metrics(self, running_apps: Dict[str, Tuple[str, str]]) -> I
             try:
                 base_url = get_request_url(self.master_address, tracking_url)
                 executors = rest_request_to_json(base_url, SPARK_APPS_PATH, app_id, "executors")
+                number_of_executors = len(executors) - 1  # Spark reports the driver as an executor, we discount it.
+                self.logger.debug(
+                    "Number of executors in app is known",
```
Comment (Jongy) [inline granulate_utils/metrics/spark.py:285]:
```suggestion
                    "Number of executors in app",
```

File: granulate_utils/metrics/spark.py (hunk 2 above)
Comment (roi-granulate) [reply granulate_utils/metrics/spark.py]: ee3d79a

File: granulate_utils/metrics/spark.py (hunk 1 above)
Comment (roi-granulate) [reply granulate_utils/metrics/spark.py]: 0a4f663

Comment (roi-granulate) [reply granulate_utils/metrics/spark.py]: comment dup

File: granulate_utils/metrics/spark.py [hunk 4]
```diff
@@ -121,8 +121,15 @@ def _get_standalone_apps(self) -> Dict[str, Tuple[str, str]]:
         # https://github.com/apache/spark/blob/67a254c7ed8c5c3321e8bed06294bc2c9a2603de/core/src/main/scala/org/apache/spark/deploy/JsonProtocol.scala#L202
         metrics_json = rest_request_to_json(self._master_address, SPARK_MASTER_STATE_PATH)
         running_apps = {}
+        current_running_apps = []
```
Comment (Jongy) [inline granulate_utils/metrics/spark.py:124]: What is `current_running_apps`? Maybe `activeapps` because that's the JSON name?

**Review by Jongy — APPROVED** [review]

**Pull request merged**